appmap/_implementation/generation.py

```diff
@@ -0,0 +1,117 @@
+"""Generate an AppMap"""
+import json
+from collections.abc import MutableMapping
+
+from . import metadata
+
+from .event import Event, serialize_event
+
+
+class ClassMapSet(MutableMapping):
+    def __init__(self):
+        self.dict = dict()
+
+    def __delitem__(self, k):
+        del self[k]
+
+    def __getitem__(self, k):
+        return self.dict[k]
+
+    def __setitem__(self, k, v):
+        self.dict[k] = v
+
+    def __iter__(self):
+        return self.dict.__iter__()
+
+    def __len__(self):
+        return len(self.dict)
+
+
+class ClassMapEntry:
```

Comment:
Also check https://docs.python.org/3/library/dataclasses.html

Reply:
Done.

Reply:
Well, actually, I tried to switch to dataclasses, but I didn't like the end result. Unless you object, I'll leave these as regular classes.